extensions/amp-story/0.1/bookend.js

```diff
@@ -209,5 +301,48 @@ export class Bookend {
     this.assertBuilt_();
     return dev().assertElement(this.root_);
   }
-}
 
+  /**
+   * @param {!AmpDoc} ampdoc
+   * @return {{
+   *   title: string,
+   *   domainName: string,
+   *   imageUrl: (string|undefined),
+   * }}
+   * @private
+   */
+  getStoryMetadata_(ampdoc) {
+    const jsonLd = getJsonLd(this.win_.document);
```

Comment:
Let's use `AmpDoc#getBody()` for proper shadow DOM embed support.

Reply:
Done